definitions/features/upstream_repositories.rb

```diff
@@ -0,0 +1,62 @@
+class Features::UpstreamRepositories < ForemanMaintain::Feature
+  metadata do
+    label :upstream_repositories
+
+    confine do
+      !feature(:instance).downstream?
+    end
+  end
+  attr_accessor :version
+
+  VERSION_MAPPING = {
+    '3.16' => '2.1',
+    '3.15' => '2.0',
+    '3.14' => '1.24'
+  }.freeze
+
+  def setup_repositories(version)
+    @version = version
+    if feature(:katello)
+      major_or_minor_repo_update(feature(:katello))
+      @version = VERSION_MAPPING[version]
```

Comment:
overriding a global variable seems dangerous.

cant we pass the right version to `major_or_minor_repo_update`?